main.cpp

```diff
@@ -0,0 +1,169 @@
+#include <string>
+#include <vector>
+#include <iostream>
+#include <mpi.h>
+
+using namespace std;
+
+#include "nnpf.h"
+
+int main(int argc, char **argv){
+
+	MPI_Init(&argc, &argv);
+
+	string load_domain_filename = "20x20rand25.dat";
+	int seeker_num = 50;
+	int runtime = 100;
+	int iterations = 200;
+	int gridx = 20;
+	int gridy = 20;
+
+
+
+	Domain domain (1,1);
+	domain.load_domain("20x20rand25.dat");
+	//domain.print_domain();
+
+
+	int size, rank;
+	MPI_Comm_size(MPI_COMM_WORLD, &size);
+	MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+	
+
+	vector<Seeker> seekers;
+	for(int i=0;i<(seeker_num/size);i++){
+		seekers.push_back(Seeker(domain.domain));
+	}
+	for (int iter=0;iter<iterations;iter++){
+		cout << "Time " << iter << " of " << iterations << endl;
+
+		string pos_filename;
+		for(int time=0;time<runtime;time++){
+			for(int i=0;i<seekers.size();i++){
+				seekers[i].move_seeker(domain.domain);
+				seekers[i].check_collisions(domain.domain);
+				seekers[i].calc_score(gridx-2,gridy-2);
+				pos_filename = "PositionalData/pos_procseeker"+to_string(1000*rank+i)+"_Iter"+to_string(iter)+".dat";
+				seekers[i].write_each_position(pos_filename);
+			}
+		string filename;
+		}
+
+		/*for (int i=0;i<seekers.size();i++){
+			pos_filename = "PositionalData/pos_procseeker"+to_string(1000*rank+i)+"_Iter"+to_string(iter)+".dat";
+			seekers[i].write_position(pos_filename);
+		}*/
+		string filename;
+		for(int i=0;i<seekers.size();i++){
+			filename = "SeekerData/SeekerPos_Core"+to_string(rank)+"_Seeker"+to_string(i)+".dat";
+			seekers[i].write_seeker(filename);
+		}
+
+		if(rank != 0){
+			double message[4];
+			for (int i=0;i<seekers.size();i++){
+				/*vector<double> message;
+				message.push_back(seekers.at(i).get_score());
+				message.push_back((double) rank);
+				message.push_back((double) i);
+				message.push_back(0.); // flag to mutate or reset, will be changed by 0 proc*/
+				
+				message[0] = seekers.at(i).get_score();
+				message[1] = (double) rank;
+				message[2] = (double) i;
+				message[3] = 0.;
+				int flag = 1000*rank+i;
+				MPI_Send(&message,4,MPI_DOUBLE,0,flag,MPI_COMM_WORLD);
```

Comment:
That's how people (including myself) typically handle sending composite messages. It's a bit Fortran-ish, but oh well ;)